src/control/run_go_tests.sh

```diff
@@ -117,18 +117,48 @@ function setup_environment()
 	fi
 }
 
+function emit_junit_failure()
+{
+    cname="run_go_tests"
+    tname="${1:-subtest}"
+    fname="${DAOS_BASE}/test_results/${cname}.${tname}.xml"
+
+    teststr="    <testcase classname=\"$cname\" name=\"$tname\">
+    <failure type=\"format\">
+      <![CDATA[$2
+        ]]>
+    </failure>
+    </testcase>"
+
+    cat > "${fname}" << EOF
+<?xml version="1.0" encoding="UTF-8" ?>
+<testsuites>
+  <testsuite tests="1" failures="1" errors="0" skipped="0" >
+EOF
+echo "${teststr}" >> "${fname}"
+cat >> "${fname}" << EOF
```

Comment:
Should be able to to use `${teststr}` in the heredoc directly:

```suggestion
${teststr}
```